### PR TITLE
encode invalid chars in version for xrefmap filename

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 1]
```diff
@@ -308,6 +307,24 @@ private static string ExportXRefMap(DocumentBuildParameters parameters, Document
...
 
+        private static string GetXrefMapFileNameWithVersion(string version)
+        {
+            if (string.IsNullOrEmpty(version))
+            {
+                return XRefMapFileName;
+            }
+
+            // TODO: full check of invalid chars
+            version = Regex.Replace(
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: what about `Uri.EscapeDataString`?

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: what for? should JSON serializer handle the case?

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: In monikerrange, write to `> netcore1.0.xrefmap.yml` will fail... Exception is thown by `Path.Combine` before passed into serializer.

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 2]
```diff
@@ -308,6 +306,16 @@ private static string ExportXRefMap(DocumentBuildParameters parameters, Document
...
         }
 
+        private static string GetXrefMapFileNameWithVersion(string version)
+        {
+            if (string.IsNullOrEmpty(version))
+            {
+                return XRefMapFileName;
+            }
+
+            return Uri.EscapeDataString(version) + "." + XRefMapFileName;
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: As it is a file name, should we use Path.GetInvalidFileNameChars?

**Pull request merged**